cc/testing/evaluation/algorithms_with_insufficient_noise.cc

```diff
@@ -0,0 +1,187 @@
+#include "algorithms/bounded-mean.h"
+#include "algorithms/count.h"
+#include "algorithms/numerical-mechanisms.h"
+#include "algorithms/numerical-mechanisms-testing.h"
+#include "algorithms/util.h"
+#include "base/statusor.h"
+#include "proto/data.pb.h"
+#include "testing/sequence.h"
+#include "testing/stochastic_tester.h"
+
+namespace differential_privacy {
+
+namespace testing {
+
+// Runs the Stochastic Tester on a continuous range of three algorithms (Count, 
+// BoundedSum, and BoundedMean) in which the privacy protection claimed exceeds 
+// the amount of noise applied. Sends results of each run to an output file.
+
+bool RunStochasticTester(std::string algorithm, double ratio, double num_datasets,
+  double num_samples_per_histogram) {
+
+  auto sequence = absl::make_unique<HaltonSequence<double>>(
+    DefaultDatasetSize(), true, DefaultDataScale(), DefaultDataOffset());
+
+  if (algorithm == "count") {
+
+    auto sequence = absl::make_unique<HaltonSequence<int64_t>>(
+      DefaultDatasetSize(), true, DefaultDataScale(), DefaultDataOffset());
+
+    auto algorithm = absl::make_unique<CountWithInsufficientNoise<int64_t>>(
+      std::log(3), ratio);
+    
+    StochasticTester<int64_t> tester( std::move(algorithm), std::move(sequence),
+      num_datasets, num_samples_per_histogram);
+
+    bool algo_is_dp = tester.Run(); // false = 0, true = 1
+    std::cout << algo_is_dp << std::endl; 
+    return algo_is_dp; // correct outcome is false, e.g., DP premise rejected
+  }
+
+  else if (algorithm == "boundedsum") {
+
+    auto algorithm = absl::make_unique<SumWithInsufficientNoise<double>>(
+      std::log(3), sequence->RangeMin(), sequence->RangeMax(),
+      absl::make_unique<test_utils::SeededLaplaceMechanism::Builder>(), ratio);
+
+    StochasticTester<double> tester(std::move(algorithm), std::move(sequence),
+      num_datasets, num_samples_per_histogram);
+
+    bool algo_is_dp = tester.Run(); // false = 0, true = 1
+    std::cout << algo_is_dp << std::endl; 
+    return algo_is_dp; // correct outcome is false, e.g., DP premise rejected
+
+  }
+
+  else if (algorithm == "boundedmean") {
+
+    auto algorithm = absl::make_unique<MeanWithInsufficientNoise<double>>(
+      std::log(3),sequence->RangeMin(), sequence->RangeMax(),
+      absl::make_unique<test_utils::SeededLaplaceMechanism::Builder>(), ratio);
+
+    StochasticTester<double> tester(std::move(algorithm), std::move(sequence),
+      num_datasets, num_samples_per_histogram);
+
+    bool algo_is_dp = tester.Run(); // false = 0, true = 1
+    std::cout << algo_is_dp << std::endl; 
+    return algo_is_dp; // correct outcome is false, e.g., DP premise rejected
+  }
+
+  else {
+    std::cout << "The algorithm specified was not recognized." << 
+      "Try count, boundedsum, or boundedmean instead." << std::endl;
+  }
+}
+
+
+void GetTestResults(std::ofstream& datafile, std::string algorithm,
+  double num_datasets, double num_samples_per_histogram, double ratio_min,
```

Comment:
Why are some of these doubles? For example, is it logically possible for num_datasets to be 2.5?

Same comment for num_tests and so on below.